lib/ansible/module_utils/vmware.py

```diff
@@ -828,41 +831,181 @@ def is_vcenter(self):
         elif api_type == 'HostAgent':
             return False
 
+    def get_managed_objects_properties(self, vim_type, properties=None):
+        """
+        Function to look up a Managed Object Reference in vCenter / ESXi Environment
+        :param vim_type: Type of vim object e.g, for datacenter - vim.Datacenter
+        :param properties: List of properties related to vim object e.g. Name
+        :return: local content object
+        """
+        # Get Root Folder
+        root_folder = self.content.rootFolder
+
+        if properties is None:
+            properties = ['name']
+
+        # Create Container View with default root folder
+        mor = self.content.viewManager.CreateContainerView(root_folder, [vim_type], True)
+
+        # Create Traversal spec
+        traversal_spec = vmodl.query.PropertyCollector.TraversalSpec(
+            name="traversal_spec",
+            path='view',
+            skip=False,
+            type=vim.view.ContainerView
+        )
+
+        # Create Property Spec
+        property_spec = vmodl.query.PropertyCollector.PropertySpec(
+            type=vim_type,  # Type of object to retrieved
+            all=False,
+            pathSet=properties
+        )
+
+        # Create Object Spec
+        object_spec = vmodl.query.PropertyCollector.ObjectSpec(
+            obj=mor,
+            skip=True,
+            selectSet=[traversal_spec]
+        )
+
+        # Create Filter Spec
+        filter_spec = vmodl.query.PropertyCollector.FilterSpec(
+            objectSet=[object_spec],
+            propSet=[property_spec],
+            reportMissingObjectsInResults=False
+        )
+
+        return self.content.propertyCollector.RetrieveContents([filter_spec])
+
     # Virtual Machine related functions
     def get_vm(self):
-        vm = None
-        match_first = (self.params['name_match'] == 'first')
+        """
+        Function to find unique virtual machine either by UUID or Name.
+        Returns: virtual machine object if found, else None.
+
+        """
+        vm_obj = None
+        user_desired_path = None
 
         if self.params['uuid']:
-            vm = find_vm_by_id(self.content, vm_id=self.params['uuid'], vm_id_type="uuid")
-        elif self.params['folder'] and self.params['name']:
-            vm = find_vm_by_id(self.content, vm_id=self.params['name'], vm_id_type="inventory_path",
-                               folder=self.params['folder'], match_first=match_first)
+            vm_obj = find_vm_by_id(self.content, vm_id=self.params['uuid'], vm_id_type="uuid")
 
-        if vm:
-            self.current_vm_obj = vm
+        elif self.params['name']:
+            objects = self.get_managed_objects_properties(vim_type=vim.VirtualMachine, properties=['name'])
+            vms = []
 
-        return vm
+            for temp_vm_object in objects:
+                for tem_vm_property in temp_vm_object.propSet:
```

Comment:
@Akasurde Is it possible to have `len(temp_vm_object.propSet) != 1` ?
Has I understand it temp_vm_object.propSet will only contain the name of the temp_vm_object.

Reply:
Yes. temp_vm_object.propSet will only contain one value. So I think there is extra for which is not required.